cEpiabm/src/sweeps/spatial_sweep.cpp

```diff
@@ -16,30 +18,68 @@ namespace epiabm
 
     void SpatialSweep::operator()(const unsigned short timestep)
     {
+  
         LOG << LOG_LEVEL_DEBUG << "Beginning Spatial Sweep " << timestep;
-        if (m_population->cells().size() <= 1){
-            return;  // no intercell infections if only one cell
+        if (m_population->cells().size() <= 1)
+        {
+            return; // no intercell infections if only one cell
         }
         m_population->forEachCell(
             std::bind(&SpatialSweep::cellCallback, this, timestep, std::placeholders::_1));
         LOG << LOG_LEVEL_DEBUG << "Finished Spatial Sweep " << timestep;
     }
 
-    inline std::vector<Cell*> getCellsToInfect(std::vector<Cell>& cells, Cell* currentCell, size_t n)
+    inline std::vector<double> getWeightsFromCells(std::vector<Cell> &cells, Cell *currentCell,
+                                                   bool doDistance = true, bool doCovidsim = false)
     {
-        std::vector<size_t> allCells = std::vector<size_t>();
-        allCells.reserve(cells.size()-1);
-        for (size_t i = 0; i < cells.size(); i++)
-            if (i != currentCell->index()) allCells.push_back(i);
+        std::vector<double> weightVector;
+        if (doDistance)
+        {
+            std::pair<double, double> current_loc = currentCell->location();
+            for (Cell &cell : cells)
+            {
+                weightVector.push_back(1./DistanceMetrics::Dist(cell.location(), current_loc));
+            }
+        }
+        // LCOV_EXCL_START
+        else if (doCovidsim) //this will be in the config file in cpp-dev
```

Comment:
Query - is `cpp-dev` a separate branch? Does that mean we need that branch in for this to work?

Reply:
doCovidsim is a true/false flag that for a simulation needs to be initialised in a config file somewhere. But everything will work with the tests just hardcoding the flag.

Reply:
Oh that makes sense - would the config file add it to a global namespace or would it need to be read in from parameters here?

Reply:
Fixed this in the cpp-dev branch, happy for this to go into main first and then we can pull in cpp-dev